Jellyfin.Server/Migrations/Routines/MigrateActivityLogDb.cs

```diff
@@ -0,0 +1,101 @@
+#pragma warning disable CS1591
+
+using System;
+using System.IO;
+using Emby.Server.Implementations.Data;
+using Jellyfin.Data.Entities;
+using Microsoft.Extensions.Logging;
+using SQLitePCL.pretty;
+
+namespace Jellyfin.Server.Migrations.Routines
+{
+    public class MigrateActivityLogDb : IMigrationRoutine
+    {
+        public Guid Id => Guid.Parse("3793eb59-bc8c-456c-8b9f-bd5a62a42978");
+
+        public string Name => "MigrateActivityLogDatabase";
+
+        public void Perform(CoreAppHost host, ILogger logger)
+        {
+            var dataPath = host.ServerConfigurationManager.ApplicationPaths.DataPath;
+            using (var connection = SQLite3.Open(
+                Path.Combine(dataPath, "activitylog.db"),
+                ConnectionFlags.ReadOnly,
+                null))
+            {
+                logger.LogInformation("This may take a while, do not stop Jellyfin.");
+                using var dbContext = host.DatabaseProvider.GetConnection();
+
+                var queryResult = connection.Query("SELECT * FROM ActivityLog ORDER BY Id ASC");
+
+                // Make sure that the database is empty in case of failed migration due to power outages, etc.
+                dbContext.Database.EnsureDeleted();
+                dbContext.Database.EnsureCreated();
```

Comment:
I think this is incorrect, the [documentation for this method](https://docs.microsoft.com/en-us/dotnet/api/microsoft.entityframeworkcore.infrastructure.databasefacade.ensurecreated?view=efcore-3.1) says

> Note that this API does not use migrations to create the database. In addition, the database that is created cannot be later updated using migrations. If you are targeting a relational database and using migrations, you can use the DbContext.Database.Migrate() method to ensure the database is created and all migrations are applied. 

I think what we need to do is define an initial migration that creates the schema and then call `Database.Migrate()` somewhere in application startup (maybe in `Program.cs` or `ApplicationHost.InitializeServices()`)

Reply:
Interesting. I didn't see the part about migrations not working on databases created using this method. That is a good catch and definitely needs to be fixed before this is merged

Reply:
Yeah, I'm very glad I caught this, that would have been quite a pain if we had initialized the database wrong for everyone on nightly hahah 😅